docs/ref/contrib/postgres/aggregates.txt

```diff
@@ -152,6 +153,62 @@ General-purpose aggregation functions
         and will be removed in Django 5.0. If you need it, explicitly set
         ``default`` to ``Value('[]')``.
 
+    Usage example::
+
+        class Room(models.Model):
+            number = models.IntegerField(unique=True)
+
+        class HotelReservation(PostgreSQLModel):
+            room = models.ForeignKey('Room', on_delete=models.CASCADE)
+            datespan = DateRangeField()
+            start = models.DateTimeField()
+            end = models.DateTimeField()
+            cancelled = models.BooleanField(default=False)
+            requirements = models.JSONField(blank=True, null=True)
+
+        >>> room101 = Room.objects.create(number=101)
+        >>> room102 = Room.objects.create(number=102)
+        >>> from django.utils import timezone
+        >>> datetimes = [
+        ...    timezone.datetime(2018, 6, 20),
+        ...    timezone.datetime(2018, 6, 24),
+        ...    timezone.datetime(2018, 6, 28),
+        ...]
+        >>> HotelReservation.objects.create(
+        ...    datespan=(datetimes[0].date(), datetimes[1].date()),
+        ...    start=datetimes[0],
+        ...    end=datetimes[1],
+        ...    room=room102,
+        ...    requirements={'double_bed': True},
+        ...)
+        >>> HotelReservation.objects.create(
+        ...    datespan=(datetimes[1].date(), datetimes[2].date()),
+        ...    start=datetimes[1],
+        ...    end=datetimes[2],
+        ...    room=room102,
+        ...    requirements={'double_bed': False, 'smoking': True},
+        ...)
+        >>> HotelReservation.objects.create(
+        ...    datespan=(datetimes[0].date(), datetimes[2].date()),
+        ...    start=datetimes[0],
+        ...    end=datetimes[2],
+        ...    room=room101,
+        ...    requirements={'smoking': False},
+        ...)
+        >>> from django.contrib.postgres.aggregates import JSONBAgg
+        >>> Room.objects.annotate(
+        ...    requirements=JSONBAgg(
+        ...        'hotelreservation__requirements',
+        ...        ordering='-hotelreservation__start',
+        ...    )
+        ...).filter(requirements__0__smoking=True).values('number', 'requirements')
+        [
+            {'number': 102, 'requirements': [
+                {'double_bed': False, 'smoking': True},
+                {'double_bed': True},
+            ]},
+        ]
```

Comment:
This example is good for tests because we already have these models, in docs we should add sth simpler, e.g. there is no need to create objects.

Reply:
We can also remove some columns like `datespan` or `cancelled`.

Reply:
> This example is good for tests because we already have these models, in docs we should add sth simpler, e.g. there is no need to create objects.

Thought so. Without creating objects, would the filtered result make sense? 🤔 

> We can also remove some columns like datespan or cancelled.

Agree 👍🏾

Reply:
> Thought so. Without creating objects, would the filtered result make sense? 

IMO the result is self-explanatory, and we don't need to initialize data in this example.

Reply:
i have removed the data creation part. in my opinion, although this simplifies it, the result could be a bit surprising for the ones reading because there is no mention of `double_bed` in the `requirements` field or in any filters, and now they would see it in the result. but we may probably keep it this way, until someone files another ticket.